src/cargo/core/shell.rs

```diff
@@ -147,27 +147,61 @@ impl MultiShell {
 }
 
 impl Shell {
-    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> Shell {
+    pub fn create(out: Box<Write + Send>, config: ShellConfig) -> CargoResult<Shell> {
```

Comment:
This is only ever really called in this one crate, so perhaps the logic of `create_fallback` could be folded into `create` itself?

That is, could we remove `create_fallback`?